pkg/agent/controller/egress/egress_controller.go

```diff
@@ -466,6 +524,31 @@ func (c *EgressController) syncEgress(egressName string) error {
 		return err
 	}
 
+	if egress.Spec.ExternalIPPool != "" && c.cluster.ShouldSelectEgress(egress) {
+		if !c.localIPDetector.IsLocalIP(egress.Spec.EgressIP) {
+			err := c.ipAssigner.AssignEgressIP(egress.Spec.EgressIP)
+			if err != nil {
+				return err
+			}
+		}
+		if err := c.updateEgressStatus(egress, c.nodeName); err != nil {
+			return fmt.Errorf("update egress status error: %v", err)
+		}
+		klog.InfoS("Assigned egress owner node", "egress", egressName, "nodeName", c.nodeName)
+	} else if egress.Status.NodeName == c.nodeName && !c.cluster.ShouldSelectEgress(egress) {
```

Comment:
So you assume the Egress IP is configured by agent when nodeName is set? I think it's better to report nodeName even for manually assigned IP, and you will need to handle the case that egress is removed during agent downtime anyway. Maybe consider using a bool in egressState to indicate whether the IP was configured by agent.

Reply:
yes,  I think your solution is more comprehensive, so we would update egressStatus whether egressIP assigned manually or automatically. And handle the IP unassignment.

Reply:
done, please help to have a look at my solution, it sames all comments has been addressed.